person.rb

```diff
@@ -0,0 +1,34 @@
+require_relative './corrector'
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @corrector = Corrector.new
+  end
+
+  def validate_name
+    @name = @corrector.correct_name(name)
```

Comment:
[OPTIONAL]
- Though the above line works, kindly note that, since `validate_name` is being passed a `name` parameter like so: validate_name(name)`, the instance variable `@name` should have been passed as a parameter to the `correct_name` method.

This 👇 is what I'm trying to say.

`@name = @corrector.correct_name(@name)`

Reply:
@DammyShittu Thank you will definitely do that